plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java

```diff
@@ -391,11 +401,20 @@ static int extractPortFromServerUrlOutput(String line) {
    * state changes.
    */
   public class DevAppServerOutputListener implements ProcessOutputLineListener {
+    // DevAppServer outputs the following for module-started and admin line (on one line):
+    // <<HEADER>> Starting module "default" running at: http://localhost:8080
+    // <<HEADER>> Starting admin server at: http://localhost:8000
+    // where <<HEADER>> = INFO 2017-01-31 21:00:40,700 dispatcher.py:197]
+    private Pattern moduleStartedPattern = Pattern.compile(
+        ".*Starting module \"(?<service>[^\"]+)\" running at: (?<url>http://.+:(?<port>[0-9]+))$");
+    private Pattern adminStartedPattern =
+        Pattern.compile(".*Starting admin server at: (?<url>http://.+:(?<port>[0-9]+))$");
 
     private int serverPortCandidate = 0;
 
     @Override
     public void onOutputLine(String line) {
+      Matcher matcher;
```

Comment:
How about declaring two matchers, one for module lines and the other for admin lines?